AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/copy-join-link/copy-join-link.component.spec.ts

```diff
@@ -0,0 +1,65 @@
+import { ElementRef } from '@angular/core';
+import { ClipboardService } from 'ngx-clipboard';
+import { CopyJoinLinkComponent } from './copy-join-link.component';
+import { ConfigService } from '../../services/config.service';
+
+describe('CopyJoinLinkComponent', () => {
+    let component: CopyJoinLinkComponent;
+    let mouseEvent: MouseEvent;
+    let clipboardServiceSpy: jasmine.SpyObj<ClipboardService>;
+    let element: HTMLDivElement;
+    let configServiceSpy: jasmine.SpyObj<ConfigService>;
+
+    const vh_video_uri = 'vh-video-web';
+
+    beforeAll(() => {
+        mouseEvent = document.createEvent('MouseEvent');
+        mouseEvent.initMouseEvent('mouseover', true, true, window, 0, 0, 0, 80, 20, false, false, false, false, 0, null);
+        clipboardServiceSpy = jasmine.createSpyObj<ClipboardService>('ClipboardService', ['copyFromContent']);
+        clipboardServiceSpy.copyFromContent.and.returnValue(true);
+
+        configServiceSpy = jasmine.createSpyObj<ConfigService>('ConfigService', ['getConfig']);
+        configServiceSpy.getConfig.and.returnValue(vh_video_uri);
+    });
+
+    beforeEach(() => {
+        component = new CopyJoinLinkComponent(clipboardServiceSpy, configServiceSpy);
```

Comment:
As we are testing the template file here, should we not hook up the angular testbed here? I can also see that we are creating an element below by using the document.createElement which shouldn't be necessary. 
Once the testbed is setup, you should be able to use the debugelement of the complied component to assert your changes? Or is there any complication when using testbed here?

Reply:
TBH, we are re-using the existing unit test so other possibilities I am sure it can be implemented